modules/dag-l1/src/main/scala/org/tessellation/dag/l1/domain/transaction/TransactionStorage.scala

```diff
@@ -35,7 +36,8 @@ import org.typelevel.log4cats.slf4j.Slf4jLogger
 
 class TransactionStorage[F[_]: Async, T <: Transaction: Order: Ordering](
   lastAccepted: MapRef[F, Address, Option[LastTransactionReferenceState]],
-  waitingTransactions: MapRef[F, Address, Option[NonEmptySet[Hashed[T]]]]
+  waitingTransactions: MapRef[F, Address, Option[NonEmptySet[Hashed[T]]]],
+  currencyIdentifier: Option[Hash]
```

Comment:
Can it ever be empty? My understanding is that it isn't ever empty. Needs to be specified for both DAG and Currency

Reply:
On one hand I think it is needed here to return lastRef on the endpoint, but I think I would rather implement this functionality (or the remaining part) as a configurable validator - which is actually needed on CL0, GL0. Interesting how hard it's gonna be to implement for GLO 🤔

Reply:
@TheMMaciek we reuse the same implementation for both DAG and Currency, hence optional type

Reply:
The hash should be required for DAG in my opinion as well. For DAG it will always be an "empty" Hash.

Reply:
@marcinwadon if we reuse the implementation then I suggest composing it.

We could create a DAG-specific implementation that encapsulates generic implementation with hardcoded `None` (or `Hash.empty`). Otherwise, it may be confusing for DAG. Mainly that `currencyIdentifier` is a public field of that storage and should not be even exposed for DAG. WDYT?

Reply:
I would just not make it Optional - for DAG pass empty hash and for currencies pass the values derived from identifier. The name can be adjusted to not suggest it is specific to currency. Maybe `initialTransactionHash`